### ISSUE TITLE
Unit Tests: Need new/updated Unit Tests for recent API changes

### ISSUE THREAD

**Issue by jburns131** [OWASP/rbac#27]
This Issue will be used to keep track of new/altered unit tests needed for v2.0 Stable release.
### Tests:
- [COMPLETE] BaseRbac::pathId() - Extended group_concat max char limit
- [COMPLETE] BaseRbac::addPath() - Should return number of nodes created
- [COMPLETE] BaseRbac::assign() - Should be able to assign using Id, Title and Path
- [COMPLETE] BaseRbac::unassign() - Should be able to unassign using Id, Title and Path
- [COMPLETE] RbacUserManager::unassign() - Should be able to unassign using Id, Title and Path

The following methods need to be changed  because addPath returns number of nodes created, not the id of the last node created:
- [COMPLETE] RbacManagerTest::testManagerCheckPath() - RbacManagerTest.php, line 143
- [COMPLETE] RbacManagerTest::testManagerEnforcePath() - RbacManagerTest.php, line 221
- [COMPLETE] RbacUsersTest::testUsersAssignWithPath() - RbacUsersTest.php, line 53
- [COMPLETE] RbacUsersTest::testUsersHasRolePath() - RbacUsersTest.php, line 122
### Results:

**Windows 8.1**
- MySQL
  - All tests pass
- SQLite
  - All tests pass

**Linux: CENTOS 6.5 x86_64**
- MySQL
  - All tests pass
- SQLite
  - All tests pass
### Notes:
- Had to change all instances of 'is_int' to 'is_numeric'
- Removed all 'core' Unit Tests
- Should we also create Unit Tests for MySQLi adapter as well? If so I'll create a new Issue to address this.
  - We might want to wait and see if we switch to a lightweight Database Abstraction Layer.
    
    I have created MySQLi specific PHPUnit connection scripts locally and there were fails and errors, but that is mainly due to the PHP MySQLi library returning strings instead of integers, and returning '-1' instead of 0 on failed queries.
    
    These return values do not seem to have a negative effect when working with the 'mysqli' adapter. The reason the tests fail is because the PHPUnit assertions test for value AND type, while PHP itself is a dynamic language and has no problems working with numeric strings vs integers.
    
    If we wanted to use a lightweight DAL while refactoring the core into PSR compliant code then the results will be consistent between databases. This is not a small decision and will take a lot of conversation and benchmarks to come to a decision.

**Comment by jburns131** [comment]
I believe all Unit Tests are complete!

Please see the comment regarding MySQLi specific tests in the 'Notes:' section of this Issue.

If we agree to wait on MySQLi specific tests AND Issue #26 is complete we can close this Issue.

**Comment by abiusx** [comment]
PHP dropped support for raw mysql interface, theres only mysqli and pdo_mysql interface, both of which should work similarly on prepared statements and queries, so no need for that atm.

**Comment by jburns131** [comment]
My thoughts also. Closing Issue.
